tensorflow/java/src/test/java/org/tensorflow/OperationBuilderTest.java

```diff
@@ -149,6 +151,36 @@ public void setAttrShape() {
     }
   }
 
+  @Test
+  public void addControlInput() {
+    assertEquals(0, lessEqualControl(2, 3).size());
+    try {
+      lessEqualControl(3, 2);
```

Comment:
There is a lot happening in lessEqualControl to be able to tie the exception to the specific line we want.
It might be better to create a simpler graph with a placeholder and have the `try{}` block around just the session run call. Something like:

```java
public void addControlInput() {
  try (Graph g = new Graph();
         Session s = new Session(g);
         Tensor yes = Tensor.create(true);
         Tensor no = Tensor.create(false)) {
    Output placeholder = TestUtil.placeholder(g, "boolean", DataType.BOOL);
    Output assert = g.opBuilder("Assert", "assert")
       .addInput(placeholder)
       .addInputList(new Output[]{})
       .build();
      Output noop = g.opBuilder("NoOp", "noop")
          .addControlInput(assert)
          .build();

     // No problems when the assert succeeds
     s.runner().feed(placeholder, yes).addTarget(noop).run();

     // Exception thrown by the execution of the Assert node
     try {
        s.runner().feed(placeholder, no).addTarget(noop).run();
        fail(...);
      } catch (IllegalArgumentException e) {
        // expected
      }
  }
}
```